README.md

```diff
@@ -162,25 +162,47 @@ Here's an example of a basic ProfileUploader component, demonstrating how to add
 
 We currently re-expose withWebId and LogoutButton so you can use the basic components without installing other libraries.
 
-### ShExFormBuilder
+### FormModel
 
-**Note**: This is an early preview build of a work-in-progress component and subject to large-scale changes in future releases.
-
-This component allows you to create a fully functional form, complete with submit, cancel, and validation rules, from a [ShEx](http://shex.io/shex-primer/index.html) Shape.
+This component takes a FormModel (build using the Form Language and [UI ontology](https://www.w3.org/ns/ui#)). A properly formatted FormModel will render a form, complete with constraints. Some things are not yet supported, such as ui:arc links, FormForms, and more. This component is still being improved and worked on.
 
 #### Supported Field Types:
 
 ##### Textbox
 
-Textboxes are the default input type for each field in a shape. In future versions, there will be a way to determine if you want a textbox vs a textarea based on properties such as maxlength.
+Textboxes are the default input type for each field in a shape, and represented in the FormModel by SingleLineTextFields. It can have common constraints like maxLength, as defined in the Form Language
+
+##### Textarea
+
+A Textarea is represented in the FormModel by a MultiLineTextField, and similar to textbox has constraints defined by the Form Language
 
 ##### Select / Dropdown
 
-Dropdowns are displayed when the shape has a reference to another shape that is only a list of values. The values are displayed in the dropdown.
+Dropdowns are displayed based on the Classifier field in the FormModel. Currently, the control will load a list of items from a Class, loading the subclasses as options in the dropdown. An example of this is the Types for Email and Phone number in the userprofile form mode.
+
+##### Date / Time / DateTime picker
+
+Using the react-datetime library, the appropriate picker is displayed according to the Form Model. There are several types of constraints, including max/minValue and a valid date range offset from "today".
+
+##### Color Picker
+
+Using the react-color library, a color picker is displayed when appropriate, saving the value as a six-digit RGB hex value.
+
+##### Integer / Decimal / Float field
+
+Numbers are represented in the form as a textbox with special custom constraints in place, enforcing that the textbox is an integer, float, or decimal depending on the Form Model definition.
+
+##### Checkbox
+
+A checkbox is rendered by a BooleanField in the Form Model, representing a binary yes/no choice. Tri-state checkboxes are not yet supported.
+
+##### Email / Phone field
+
+Email and Phone Number fields are custom textboxes, similar to the Number fields. They each take a regex pattern from the form model and
```

Comment:
Missing end of sentence "... from the form model and" And what? I'm left in suspense...

Reply:
Always keep them wanting more